pkg/util/computeskus/computeskus.go

```diff
@@ -0,0 +1,32 @@
+package computeskus
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	mgmtcompute "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2020-06-01/compute"
+)
+
+// Zones returns zone information for the resource SKU
+func Zones(sku *mgmtcompute.ResourceSku) []string {
+	if sku.LocationInfo == nil || len(*sku.LocationInfo) == 0 {
+		return nil
+	}
+
+	return *(*sku.LocationInfo)[0].Zones
```

Comment:
I might be paranoid, but should not we test the "nillity" of `(*sku.LocationInfo)[0].Zones` before dereferencing it ?

Reply:
`len(*sku.LocationInfo) == 0` above makes sure that `*sku.LocationInfo` doesn't point to `nil` (`len()` can take `nil`) and not empty, so there should be no `nil` dereference error.

As for checking whether `*(*sku.LocationInfo)[0].Zones` is `nil` or not - it should be a concern of the calling code.

Or did I misunderstood something?

Reply:
For visibility. Had a chat with Ghislain about it. Here is what I missed: `*sku.LocationInfo[0].Zones` is a `*[]string`, There will be dereferencing error in `*(*sku.LocationInfo)[0].Zones`, if it is `nil`.

I'll add an extra check and add a unit test case for it shortly.

Reply:
Updated.